core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java

```diff
@@ -204,31 +211,22 @@ public E match(String potentialMatch) {
      * @param vars A Map of wildcard-matched strings
      * @return The new value
      */
-    protected String convertParam(String val, Map<String, String> vars) {
-        if (val == null) {
-            return null;
-        } 
-        
-        int len = val.length();
-        StringBuilder ret = new StringBuilder();
-        char c;
-        String varVal;
-        for (int x=0; x<len; x++) {
-            c = val.charAt(x);
-            if (x < len - 2 && 
-                    c == '{' && '}' == val.charAt(x+2)) {
-                varVal = vars.get(String.valueOf(val.charAt(x + 1)));
-                if (varVal != null) {
-                    ret.append(varVal);
-                } 
-                x += 2;
-            } else {
-                ret.append(c);
-            }
-        }
-        
-        return ret.toString();
-    }
+	protected String convertParam(String val, Map<String, String> vars) {
+		if (val == null) {
+			return null;
+		}
+
+		Pattern wildcardPattern = Pattern.compile("\\{(.)\\}");
```

Comment:
Just a small optimisation, could you move this pattern compilation into a field? Right now each time the `convertParam` is called, the same pattern is compiled over and over.

Reply:
Moved.

Reply:
👏